src/agent/mod.rs

```diff
@@ -434,6 +434,8 @@ where
         let client = self.client.as_ref().ok_or(OAuth2Error::NotInitialized)?;
         let config = self.config.as_ref().ok_or(OAuth2Error::NotInitialized)?;
 
+        let post_login_url = Self::current_url().map_err(OAuth2Error::StartLogin)?;
+
         // take the parameter value first, then the agent configured value, then fall back to the default
         let redirect_url = match options.redirect_url.or_else(|| {
```

Comment:
You can now do a bit of a deduplication here, because match expression made sense mostly for lazy evaluation of current_url getter. (I'd personally also rename `post_login_url` to `current_url` to keep naming consistent with current context and for deduplication to make more sense if you go for it, but that's more a matter of taste :) )

```rust
        let redirect_url = options.redirect_url.or_else(|| {
            config
                .options
                .as_ref()
                .and_then(|opts| opts.redirect_url.clone())
        }).unwrap_or(post_login_url);
```